CMS-Enterprise/sbom-harbor-ui
# feat(components): add MultiDropzone component

## Summary

- [x] ~_⚠️ depends on #90, do not merge until #90 has been merged and this has been rebased._~

This PR introduces a new feature, `MultiDropzone`, a drag-and-drop component that also supports file selection for file uploads. It's designed to handle multiple file types and offers a range of customizability options to suit various use cases.

### Added

- Created a new `MultiDropzone` component that provides an area for users to drag and drop files or select files for upload. The `MultiDropzone` component includes support for multiple files, file size restrictions, and file type restrictions. It handles displaying the status of each file upload and allows for custom error messages.
- The `MultiDropzone` utilizes utility functions to format accepted file types, extract error messages, and manage file upload status.
- Added a new `UploadFileCell` component that represents each file that is selected for upload within the `MultiDropzone`.
- Constants and types that are relevant to the `MultiDropzone` and `UploadFileCell` components have been defined.
- Storybook stories to accompany the components.

### Changed

- n/a

### Deprecated

- n/a

### Removed

- n/a

### Fixed

- n/a

## How to test

- `yarn` install new dependencies
- `yarn test` ensure all tests pass
- `yarn storybook` start storybook
- to try the dropzone component, navigate to http://localhost:6006/?path=/story/upload-ui-components-multidropzone--playground

Final codecoverage

```
-------------------------------------|---------|----------|---------|---------|-------------------
File                                 | % Stmts | % Branch | % Funcs | % Lines | Uncovered Line #s
-------------------------------------|---------|----------|---------|---------|-------------------
 src/components/MultiDropzone        |   95.74 |     85.1 |   88.46 |   99.02 |
  MultiDropzone.tsx                  |   91.11 |    71.11 |   76.92 |    97.5 | 197
  UploadFileCell.tsx                 |     100 |    96.29 |     100 |     100 | 130
  constants.ts                       |      90 |      100 |     100 |     100 |
  types.ts                           |     100 |      100 |     100 |     100 |
  utils.ts                           |     100 |      100 |     100 |     100 |
```

## Screenshots

- ![image](https://github.com/CMS-Enterprise/sbom-harbor-ui/assets/1915925/ffc94be4-6b57-4bce-821a-f701f2e3fdaf)
